test-utils/src/main/java/com/revenuecat/purchases/utils/productStubs.kt

```diff
@@ -56,22 +57,76 @@ fun stubStoreProduct(
     override val description: String
         get() = ""
     override val period: Period?
-        get() = subscriptionOptions.firstOrNull { it.isBasePlan }?.pricingPhases?.get(0)?.billingPeriod
-    override val subscriptionOptions: SubscriptionOptions
-        get() = SubscriptionOptions(subscriptionOptions)
+        get() = subscriptionOptions?.firstOrNull { it.isBasePlan }?.pricingPhases?.get(0)?.billingPeriod
```

Comment:
the stubs here highlight a frustration i have with this approach: someone could create a StoreProduct with an offering Id, and it wouldn't automatically propagate to the subscription options.

so you could get into an "invalid" state. i can't think of a non-hacky way to do this..

Reply:
we could have separate version of the options properties called something like "validSubscriptionOptions" that are computed properties, combining `subscriptionOptions` with the `presentedOfferingId`..

seems like overkill though